src/App.js

```diff
@@ -2,24 +2,94 @@ import React from 'react';
 
 import './App.scss';
 
-const App = () => {
-  setInterval(() => {
-    const date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
-};
+class App extends React.Component {
+  state = {
+    time: new Date(),
+    isClockVisible: true,
+    clockName: Math.floor(Math.random() * (10 - 1)) + 1,
+  }
+
+  componentDidMount() {
+    this.timerID = setInterval(() => {
+      if (this.state.isClockVisible) {
+        this.tick();
+        // eslint-disable-next-line
+        console.log(this.state.time.toLocaleTimeString());
+      }
+    }, 1000);
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevState.clockName !== this.state.clockName) {
+      // eslint-disable-next-line
+      console.log(`The Clock was renamed from ${prevState.clockName} to ${this.state.clockName}`);
+    }
+  }
```

Comment:
It's great that you have used componentDidUpdate, but for this task you should only console.log the new name only when clicking the button (so you could easily put the 'console.log to your 'randomName' method)

Reply:
Okey